src/store/registerUser.js

```diff
@@ -0,0 +1,35 @@
+import { defineStore } from 'pinia';
+import { getAuth, createUserWithEmailAndPassword } from "firebase/auth";
+import { collection, addDoc } from "firebase/firestore";
+import { getStorage, ref, uploadBytes } from "firebase/storage";
+import { app, db } from '../firebase.js';
+import { reactive, toRefs } from 'vue';
+const auth = getAuth(app);
+const storage = getStorage();
+export const userRegisterUse = defineStore('userRegister', () => {
+    const state = reactive({
+        existUserError: false
+    })
+
+    const createUser = async (userData) => {
+        try {
+            const authResult = await createUserWithEmailAndPassword(auth, userData.email, userData.password);
+            const storedata = await addDoc(collection(db, "userDetails"), {
+                uid: authResult.user.uid,
+                firstName: userData.firstName,
+                lastName: userData.lastName,
+                mobileNumber: userData.mobileNumber,
+                email: userData.email
+            })
+            const storageRef = ref(storage, `userProfile/${authResult.user.uid}/profilepic`);
+            uploadBytes(storageRef, userData.profilePhoto);
```

Comment:
you also need to save profile photo path to user data otherwise how you will be going to show it? @ShivanshSinghnnovative